### ISSUE TITLE
Wrong linter being used?

### ISSUE THREAD

**Issue by OmgImAlexis** [steelbrain/linter#1574]
I have `linter-xo` installed and even though it should only match js files it's trying to lint my `HTML (mako)` files and not just the code inside the script tags.

I'm not sure if this is an issue with `linter-xo` or `linter` itself.

![screen shot 2018-02-28 at 1 01 00 pm](https://user-images.githubusercontent.com/6525926/36766972-224e0ede-1c88-11e8-8d61-f320b247e091.png)

**Comment by Arcanemagus** [comment]
It [looks like](https://github.com/xojs/atom-linter-xo/blob/53ca9ad16b4656fc4129595b4e7b83b8aa30fd79/index.js#L12) `linter-xo` marks itself as supporting embedded JavaScript. You'll have to file an issue there as to why they are claiming support for that but haven't configured the underlying ESLint instance with the proper plugin in order to actually lint that.

You would have to check with `Linter: Debug` or `Editor: Log Cursor Scopes` from the Command Palette, but I'm guessing that at some point your cursor entered an area that the `HTML (Mako)` grammar considered to be embedded JavaScript, and triggered XO. If it _is_ using `eslint-plugin-html`, I know that _only_ handles pure HTML and would likely get _very_ confused when hitting the Mako code.

The only part of this that is possibly something `linter` could change is that it only triggers providers when a scope they mark themselves as understanding is active, in this case when you _aren't_ in an embedded JavaScript section of the code `linter-xo` wouldn't even be triggered, meaning it couldn't update any results until you re-enter a matching section of the code. Atom is somewhat limited in the scopes it exposes though so the only ways to fix this would be drastic and have major performance implications.

tl;dr:
* `HTML (Mako)` should probably be marking the sections with `source.js.embedded.mako`, not `...html`
* `linter-xo` _may_ need additional configuring to handle embedded JavaScript, I'm not sure as I don't use it myself

**Comment by OmgImAlexis** [comment]
The file: https://gist.github.com/OmgImAlexis/7b378faa7fd999349dc47fb990f44557

I was expecting the js in the script tags to be linted but the script tags using `type="text/x-template"` to be skipped since it's not javascript. Maybe that's why this is having issues?

```
Platform: darwin
Atom Version: 1.24.0
Linter Version: 2.2.0
Opened file is ignored: No
Matching Linter Providers: 
Disabled Linter Providers: 
Standard Linter Providers: 
  - pycodestyle
  - stylelint
  - SwaggerParser
  - XO
Indie Linter Providers: 
UI Providers: 
  - Linter
Ignore Glob: **/*.min.{js,css}
VCS Ignored Paths are excluded: true
Current File Scopes: 
  - *
  - text.html.mako
  - meta.tag.script.html
  - text.embedded.html
  - meta.tag.inline.option.html
  - punctuation.definition.tag.begin.html
```
